dist/rpm/singularity.spec.in

```diff
@@ -84,10 +84,9 @@ export GOPATH=$PWD/gopath
 export PATH=$GOPATH/bin:$PATH
 cd $GOPATH/%{singgopath}
 
-./mconfig -V %{version}-%{release} --prefix=%{_prefix} --exec-prefix=%{_exec_prefix} \
-	--bindir=%{_bindir} --libexecdir=%{_libexecdir} --sysconfdir=%{_sysconfdir} \
-	--sharedstatedir=%{_sharedstatedir} --localstatedir=%{_localstatedir} \
-	--libdir=%{_libdir}
+./mconfig -V %{version}-%{release} \
+        --exec-prefix=%{_exec_prefix} --sysconfdir=%{_sysconfdir} \
+	--localstatedir=%{_localstatedir} --mandir=%{_mandir}
```

Comment:
Hold up here. I totally skipped over this.... Why are you removing the other values (`prefix`, `binddir`, `libexecdir`, etc...)?

I pretty much replicated the `%configure` RPM macro action that happens with `autoconf`. *That* should be the expected behavior we replicate, not this.

Reply:
Further in the spec file in `%files` we do things like:

```
%{_libexecdir}/singularity/cni/*
```

You are not overriding `_libexecdir` so the default value will be used, which may or may not be what you expect it to be based on the values you are passing in.

Reply:
I found through testing that these were the only values that were needed.  Some of the ones that were there had no effect at all, including --prefix.  In fact the default value of --prefix on rhel was /usr, which was wrong.  If we need a slightly different value for one or two of the others on suse, such as libexecdir, I could add them back.

Reply:
I think it would be best to add them back since all of the OSes have differing opinions on things. This way we know we will at least be consistent.

Reply:
On SLES, the *default* RPM value is:

```
%_libexecdir      %{_exec_prefix}/libexec
```

But a later file changes it to being:
```
%_libexecdir     %{_exec_prefix}/lib
```

`mconfig` makes the assumption that if `--libexecdir` is not specified, the value will be:  `$exec_prefix/libexec` ... But is some cases that isn't true.

Reply:
Yes, --exec_prefix took care of _libexecdir and _bindir.  I am now testing replacing the former with the latter two.  That will then be setting all 5 install directory macros used in singularity.spec.

Reply:
Can you make it as it was. What you're doing is making the assumption (no matter how sane) that no one is ever going to want to do something like:

```
rpmbuild -tb --define '_prefix /usr/local' --define '_exec_prefix /usr' --define '_bindir /binaries/' --define '_sysconfdir /configuration' singularity-[ver].tar.gz
```

Keep it as close to the normal `%configure` macro as possible.

Ohh... and on that note, I forgot we can actually add ion `--datadir`, `--mandir`, etc... to the `./mconfig` line there in the spec file.

Reply:
Passing the other defines to mconfig won't have any effect, because they are not used anywhere in singularity.spec and the 5 that are used are completely specified.   I made the commits I talked about and tested them on rhel6, rhel7, and opensuse (the latter under docker).  I also verified that setting RPMPREFIX=/usr/local works.  On opensuse, /usr/lib was used instead of /usr/libexec.  See if you can find a case where this version breaks or does something unexpected.

Reply:
@jmstover convinced me by Slack to have all of the %configure defines, in case they are used in the future.  This now also fixes issue #2636.